kiwi_lint/db_column.py

```diff
@@ -0,0 +1,25 @@
+from pylint import interfaces, checkers
+from pylint.checkers import utils
+
+
+class DbColumnChecker(checkers.BaseChecker):
+    __implements__ = (interfaces.IAstroidChecker, )
+
+    name = 'db-column-used'
+
+    msgs = {
+        'E4841': (
+            'Do not use db_column in model field definitions',
+            'db-column-used',
+            'Do not use db_column argument in model field definitions,'
+            'See: https://github.com/kiwitcms/Kiwi/issues/736'
+        )
+    }
+
+    @utils.check_messages('db-column-used')
+    def visit_attribute(self, node):
+        if (node.attrname.endswith(('Field', 'Key'))
+                and node.expr.name == 'models' and node.parent.keywords):
+            for keyword_arg in node.parent.keywords:
+                if keyword_arg.arg == 'db_column':
+                    self.add_message('db-column-used', node=node)
```

Comment:
To my knowledge only model fields accept the db_column attribute so the simplest way to check is for anything that accepts attributes to just check if the attribute is named db_column regardless of the type of the surrounding node.